node/overseer/src/lib.rs

```diff
@@ -314,6 +310,103 @@ impl<T> From<T> for MaybeTimed<T> {
+	pov_distribution: mpsc::Sender<MessagePacket<PoVDistributionMessage>>,
+	runtime_api: mpsc::Sender<MessagePacket<RuntimeApiMessage>>,
+	availability_store: mpsc::Sender<MessagePacket<AvailabilityStoreMessage>>,
+	network_bridge: mpsc::Sender<MessagePacket<NetworkBridgeMessage>>,
+	chain_api: mpsc::Sender<MessagePacket<ChainApiMessage>>,
+	collation_generation: mpsc::Sender<MessagePacket<CollationGenerationMessage>>,
+	collator_protocol: mpsc::Sender<MessagePacket<CollatorProtocolMessage>>,
+}
+
+impl ChannelsOut {
+	async fn send_and_log_error(
+		&mut self,
+		t: MaybeTimer,
+		signals_received: usize,
+		message: AllMessages,
+	) {
+		fn make_packet<T>(timer: MaybeTimer, signals_received: usize, message: T) -> MessagePacket<T> {
+			MessagePacket {
+				signals_received,
+				message: MaybeTimed { timer, t: message },
+			}
+		}
+
+		let res = match message {
+			AllMessages::CandidateValidation(msg) => {
+				self.candidate_validation.send(make_packet(t, signals_received, msg)).await
+			},
+			AllMessages::CandidateBacking(msg) => {
+				self.candidate_backing.send(make_packet(t, signals_received, msg)).await
+			},
+			AllMessages::CandidateSelection(msg) => {
+				self.candidate_selection.send(make_packet(t, signals_received, msg)).await
+			},
+			AllMessages::StatementDistribution(msg) => {
+				self.statement_distribution.send(make_packet(t, signals_received, msg)).await
+			},
+			AllMessages::AvailabilityDistribution(msg) => {
+				self.availability_distribution.send(make_packet(t, signals_received, msg)).await
+			},
+			AllMessages::BitfieldDistribution(msg) => {
+				self.bitfield_distribution.send(make_packet(t, signals_received, msg)).await
+			},
+			AllMessages::BitfieldSigning(msg) => {
+				self.bitfield_signing.send(make_packet(t, signals_received, msg)).await
+			},
+			AllMessages::Provisioner(msg) => {
+				self.provisioner.send(make_packet(t, signals_received, msg)).await
+			},
+			AllMessages::PoVDistribution(msg) => {
+				self.pov_distribution.send(make_packet(t, signals_received, msg)).await
+			},
+			AllMessages::RuntimeApi(msg) => {
+				self.runtime_api.send(make_packet(t, signals_received, msg)).await
+			},
+			AllMessages::AvailabilityStore(msg) => {
+				self.availability_store.send(make_packet(t, signals_received, msg)).await
+			},
+			AllMessages::NetworkBridge(msg) => {
+				self.network_bridge.send(make_packet(t, signals_received, msg)).await
+			},
+			AllMessages::ChainApi(msg) => {
+				self.chain_api.send(make_packet(t, signals_received, msg)).await
+			},
+			AllMessages::CollationGeneration(msg) => {
+				self.collation_generation.send(make_packet(t, signals_received, msg)).await
+			},
+			AllMessages::CollatorProtocol(msg) => {
+				self.collator_protocol.send(make_packet(t, signals_received, msg)).await
+			},
+		};
+
+		if res.is_err() {
+			tracing::debug!(
+				target: LOG_TARGET,
+				"Failed to send a message to another subsystem",
```

Comment:
I suspect that at some point we're going to want to debug which particular subsystem was the intended recipient of the failed message. Maybe an api like 

```rust
impl AllMessages {
  pub fn recipient_name(&self) -> &'static str { ... }
}
```

would make that simple to add to this debug message.